Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift

```diff
@@ -211,56 +226,51 @@ extension GeometryEditorMenu {
     ///
     /// In some instances deleting the selection may be invalid. One example would be the mid vertex
     /// of a line.
-    var deleteButtonIsDisabled: Bool {
-        guard let selection = model.selection else { return true }
-        return !selection.canBeDeleted
+    private var deleteButtonIsDisabled: Bool {
+        guard let selectedElement else { return true }
+        return !selectedElement.canBeDeleted
     }
-}
-
-/// An object that acts as a view model for the geometry editor menu.
-@MainActor
-class GeometryEditorMenuModel: ObservableObject {
-    /// The geometry editor.
-    let geometryEditor: GeometryEditor
-    
-    /// The graphics overlay used to save geometries to.
-    let graphicsOverlay: GraphicsOverlay
     
     /// A Boolean value indicating if the geometry editor can perform an undo.
-    @Published private(set) var canUndo = false
+    private var canUndo: Bool {
+        return model.geometryEditor.canUndo
+    }
     
     /// A Boolean value indicating if the geometry editor can perform a redo.
-    @Published private(set) var canRedo = false
-    
-    /// The currently selected element.
-    @Published private(set) var selection: GeometryEditorElement?
+    private var canRedo: Bool {
+        return model.geometryEditor.canRedo
+    }
     
     /// A Boolean value indicating if the geometry can be saved to a graphics overlay.
-    @Published private(set) var canSave = false
+    private var canSave: Bool {
+        return geometry?.sketchIsValid ?? false
+    }
     
     /// A Boolean value indicating if the geometry can be cleared from the geometry editor.
-    @Published private(set) var canClearCurrentSketch = false
+    private var canClearCurrentSketch: Bool {
+        return geometry.map { !$0.isEmpty } ?? false
+    }
+}
+
+/// An object that acts as a view model for the geometry editor menu.
+@MainActor
+private class GeometryEditorModel: ObservableObject {
+    /// The geometry editor.
+    let geometryEditor = GeometryEditor()
+    
+    /// The graphics overlay used to save geometries to.
+    let geometryOverlay = GraphicsOverlay(renderingMode: .dynamic)
     
     /// A Boolean value indicating if the saved sketches can be cleared.
     @Published private(set) var canClearSavedSketches = false
     
-    /// The current geometry of the geometry editor.
-    @Published private(set) var geometry: Geometry? {
-        didSet {
-            canUndo = geometryEditor.canUndo
-            canRedo = geometryEditor.canRedo
-            canClearCurrentSketch = geometry.map { !$0.isEmpty } ?? false
-            canSave = geometry?.sketchIsValid ?? false
-        }
-    }
-    
     /// A Boolean value indicating if the geometry editor has started.
-    @Published var isStarted = false
+    @Published private(set) var isStarted = false
     
     /// A Boolean value indicating if the scale mode is uniform.
     @Published var shouldUniformScale = false {
```

Comment:
Something seems off with this name.. maybe `isUniformScale`?

Reply:
Phil G had the same question [here](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/372#discussion_r1561367356)… I think it warrants a change. 😥